### ISSUE TITLE
Configure default registry in less invasive manner

### ISSUE THREAD

**Issue by kennyjwilli** [metosin/malli#488]
I'm curious if Malli has considered allowing the default registry to be set in a less invasive way (i.e., not through jvm props)? The requirement to set a jvm prop touches many places in a large application and, I imagine, will cause future developer confusion by requiring every repl to be launched with that prop.

I originally asked this question in #malli in Slack. Pulling it out here for further discussion. 

It seems like an acceptable workaround for the time being is to alter the def of `malli.registry/registry*` with your own value, avoiding the [custom check](https://github.com/metosin/malli/blob/2d3d6ff54d5df6c55a63ca46a09ab6586607f9ac/src/malli/registry.cljc#L28). 
```clojure
(alter-var-root #'malli.registry/registry* (constantly (atom registry)))
```

**Comment by ikitommi** [comment]
indeed.

**Comment by kennyjwilli** [comment]
The workaround is a bit more complicated than I originally thought. Needs to look something like this. 
```clojure
(defn init!
  [registry]
  (let [registry-atom (atom registry)]
    #?(:clj  (do
               (alter-var-root #'malli.registry/type (constantly (.intern "custom")))
               (alter-var-root #'malli.registry/registry* (constantly registry-atom))
               (alter-var-root #'m/default-registry (constantly (malli.registry/custom-default-registry))))
       :cljs (do
               (set! malli.registry/type "custom")
               (set! malli.registry/registry* registry-atom)
               (set! m/default-registry (malli.registry/custom-default-registry))))))
```

**Comment by ikitommi** [comment]
will be fixes in #583.
